### ISSUE TITLE
Rename NDFrame.filter to .select?

### ISSUE THREAD

**Issue by topper-123** [pandas-dev/pandas#26642]
I've made a PR to remove ``NDFrame.select``, see #26641. That method was deprecated in 0.21.

There was a discussion in #12401, where many participants agreed that removing the old ``select`` and having ``NDFrame.filter`` renamed to ``NDFrame.select`` would be a better naming scheme. I agree on that for several of the reasons mentioned in the thread.

What do people think of such a name change and a related deprecation of the name (not functionality) ``.filter``?

**Comment by WillAyd** [comment]
I would agree with the naming scheme. So is your proposal to deprecate both with a warning for filter that it will eventually become select?

**Comment by topper-123** [comment]
``.select`` was already deprecated in 0.21 and will be removed in 0.25. But yeah, after #26641 is merged, just change ``.filter`` to ``.select`` and make ``.filter`` a pass-through function like this:

```python
def filter(self, items=None, like=None, regex=None, axis=None):
    warnings.warn(..., FutureWarning)
    return self.select(items=None, like=None, regex=None, axis=None)
```

**Comment by topper-123** [comment]
I've made a PR on this, see #26866,

There are some things I don't like in the current signature though:
* ``like`` is very simple and matches anywhere in a string,  so I would probably never use it, in fear of accepting something wrong. Should it be taken out?
* ``.select`` currently doesn't accept a callable. A callable would be the more readable alternative to using regexes. Should .select accept callables, e.g. ``lambda x: x.startswith('abc')``? This would make ``.select`` be usable for non-string indexes also.
* both ``like`` and ``regex`` internally convert all labels to strings using ``ensure_str``.  I think it should be the users responsibility to only use regex with string labels. Should use of ``ensure_str`` be removed?

**Comment by topper-123** [comment]
Thinking a bit more: most of the functionality of ``.filter`` is already in ``.loc``. The only one missing is really regex matching. I wouldn´t be opposed to *drop* adding this ``.select`` method, but add a ``.select_regex`` method instead to complement ``.select_dtypes``, and recommend all other axis filtering being done in ``.loc`` (and ``__getitem__``).

This would be focused and we`d avoid having similar functionality in different methods.

**Comment by WillAyd** [comment]
Yea there is a ton of overlap between all of these items. I like your proposal for `.select_regex` and pushing everything else to `.loc`, so long as we don't feel like we need to remove some functionality from `.loc`

**Comment by jorisvandenbossche** [comment]
As I also commented on the PR, I personally like having more "complex" behaviour (things that need keyword arguments, like the regex) in a method like `select`, rather than further overloading `loc`
